runtime/include/gpu/cuda/chpl-gpu-gen-includes.h

```diff
@@ -26,6 +26,7 @@
 #include "chpltypes.h"
 #include "chpl-comm.h"
 
+extern __device__ c_nodeid_t chpl_nodeID;
```

Comment:
> I wonder if it's worth naming this differently (say d_chpl_nodeID or chpl_d_nodeId or chpl_dev_nodeId or something) given that its such a convention with the CUDA code I've seen to make clear what variables are on the host side and what variable are on the device (especially for variables where there's both a host and device version).

I went back and forth on this, and I choose to keep it the same as the host version based on the ground that it might be confusing if we have two different names. I won't fight too much for it, though.

> Right now we have both a host and device version of chpl_nodeID so which one you're referring to will come down to which header you include, right?
>
> I'm actually a little confused that you don't run into some kind of name collision with your extern chpl_nodeID; in gpu-cuda.c, but you got this to compile so obviously that's not happening.

With GPU locale model, we use C++ linkage. You can have overloads in C++ as opposed to C and you can also overload names based on host-ness and device-ness of them.

Reply:
> I won't fight too much for it, though.

Yeah I won't fight too much for having separate names either. I guess let's keep it "as is" for now and keep it in the back of our head as we add more globals (to see if it having separate names would clarify anything worthwhile).

> You can have overloads in C++ as opposed to C and you can also overload names based on host-ness and device-ness of them.

Really? Ok if you host-ness/device-ness (hostitude/deviceitude?) is overloadable I guess this bothers me less.